### PR TITLE
PR-609 => develop

### PR DESCRIPTION
**Pull request by nico-isento**

### CODE REVIEW

**Comment by nico-isento** [comment]
# Jira Ticket:
Link to corresponding jira ticket:
https://pib-rocks.atlassian.net/jira/software/c/projects/PR/boards/3?selectedIssue=PR-609

# Checklist:
Necessary tasks before pull request can be merged:
- [ ] All acceptance criteria have been implemented
- [ ] The code has been reviewed and approved at least once
- [ ] All necessary tests have been successfully conducted
- [ ] If necessary: A documentation covering the new concepts and changes has been created

File: src/app/shared/services/program.service.ts [hunk 1]
```diff
@@ -5,24 +5,39 @@ import {BehaviorSubject, Observable} from "rxjs";
 import {UrlConstants} from "./url.constants";
 import {ProgramCode} from "../types/program-code";
 import {UtilService} from "./util.service";
+import {RosService} from "./ros-service/ros.service";
+import {ProgramOutputLine as ProgramOutputLineRos} from "../ros-types/msg/program-output-line";
+import {ProgramOutputLine} from "../types/program-output-line";
+import {ExecutionState, ProgramState} from "../types/program-state";
+import {line} from "blockly/core/utils/svg_paths";
```
Comment (JaHe-isento) [inline src/app/shared/services/program.service.ts:12]: This import seems to be unused

File: src/app/shared/services/program.service.ts [hunk 2]
```diff
@@ -5,24 +5,39 @@ import {BehaviorSubject, Observable} from "rxjs";
 import {UrlConstants} from "./url.constants";
 import {ProgramCode} from "../types/program-code";
 import {UtilService} from "./util.service";
+import {RosService} from "./ros-service/ros.service";
+import {ProgramOutputLine as ProgramOutputLineRos} from "../ros-types/msg/program-output-line";
```
Comment (JaHe-isento) [inline src/app/shared/services/program.service.ts:9]: These imports seem to be unused

File: src/app/program/program-splitscreen/program-workspace/program-workspace.component.spec.ts [hunk 3]
```diff
@@ -71,113 +21,83 @@ describe("ProgramWorkspaceComponent", () => {
...
+        expect(codeVisualSpy).not.toHaveBeenCalled();
+        expect(codePythonSpy).not.toHaveBeenCalled();
+        expect(codePythonEmitSpy).not.toHaveBeenCalled();
     });
 
-    it("should change the viewMode to normal view if splitscreenMode is true ", () => {
-        const viewModeSubjectSpy = spyOn(
-            programService.viewModeSubject,
-            "next",
+    it("should emit the new flyout wdith if content open", () => {
```
Comment (JaHe-isento) [inline src/app/program/program-splitscreen/program-workspace/program-workspace.component.spec.ts:78]: Typo: "width" instead of "wdith"

**Review by JaHe-isento — CHANGES_REQUESTED** [review]

File: src/app/program/program-splitscreen/console/console.component.html [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    </div>
+    <div
+        *ngIf="state.executionState >= ExecutionState.STARTING"
+        class="info">
+        Starting execution of program...
+    </div>
+    <div
+        *ngIf="state.executionState === ExecutionState.NOT_STARTED"
+        class="info">
+        Program has not started yes (click 'Run' to start execution)
```
Comment (JaHe-isento) [inline src/app/program/program-splitscreen/console/console.component.html:37]: Typo: "... not started yet" instead of  "... not started yes"

**Review by JaHe-isento — APPROVED** [review]

**Pull request merged**